### PR TITLE
get exitcode from inspectExecCmd

### PR DESCRIPTION
**Pull request by dmarkhas**
Addresses #1309

### CODE REVIEW

File: core/src/main/java/org/testcontainers/containers/Container.java [hunk 1]
```diff
@@ -40,14 +40,18 @@ default SELF self() {
...
         private final String stderr;
 
-        public ExecResult(String stdout, String stderr) {
+        public ExecResult(Integer exitCode, String stdout, String stderr) {
+            this.exitCode = exitCode;
             this.stdout = stdout;
             this.stderr = stderr;
         }
 
+        public Integer getExitCode() {return  exitCode;}
```
Comment (bsideup) [inline core/src/main/java/org/testcontainers/containers/Container.java]:
```suggestion
        public Integer getExitCode() {
                return exitCode;
        }
```

File: core/src/main/java/org/testcontainers/containers/Container.java [hunk 2]
```diff
@@ -40,14 +40,18 @@ default SELF self() {
      * docker API, there's no easy way to get the result code from the process we ran.
      */
     class ExecResult {
+        private final Integer exitCode;
         private final String stdout;
         private final String stderr;
 
-        public ExecResult(String stdout, String stderr) {
+        public ExecResult(Integer exitCode, String stdout, String stderr) {
```
Comment (bsideup) [inline core/src/main/java/org/testcontainers/containers/Container.java]:
```suggestion
        public ExecResult(int exitCode, String stdout, String stderr) {
```

File: core/src/main/java/org/testcontainers/containers/Container.java [hunk 3]
```diff
@@ -40,14 +40,18 @@ default SELF self() {
      * docker API, there's no easy way to get the result code from the process we ran.
      */
     class ExecResult {
+        private final Integer exitCode;
```
Comment (bsideup) [inline core/src/main/java/org/testcontainers/containers/Container.java]:
```suggestion
        private final int exitCode;
```

**Comment by bsideup** [comment]
@dmarkhas 

FYI I've pushed a small change (Lombok's `@Value` and added a now-deprecated constructor to avoid a breaking change).

Happy to merge after the CI reports green status, good job 👍

**Comment by dmarkhas** [comment]
Thanks @bsideup , I did not find any other usages of this class in the project, and since it is not public I assumed it is safe to change the ctor signature.

**Comment by bsideup** [comment]
@dmarkhas it is public (defined as an inner class of an interface)

**Comment by dmarkhas** [comment]
Right, sorry :)

**Comment by bsideup** [comment]
@dmarkhas oh, no problem :)
That was actually an issue on our side (public constructor for a result-holding class), good that we fixed it as part of this PR 👍

**Comment by bsideup** [comment]
@dmarkhas merged, thanks! 👍

**Comment by rnorth** [comment]
Releasing this in [1.11.0](https://github.com/testcontainers/testcontainers-java/releases/tag/1.11.0) 🎉

**Pull request merged**